azurerm/resource_arm_container_group_test.go

```diff
@@ -63,24 +67,61 @@ func TestAccAzureRMContainerGroup_imageRegistryCredentialsUpdate(t *testing.T) {
 			{
 				Config: config,
 				Check: resource.ComposeTestCheckFunc(
-					testCheckAzureRMContainerGroupExists(resourceName),
+					testCheckAzureRMContainerGroupExists(resourceName, func(cg containerinstance.ContainerGroup) error {
```

Comment:
We should not be doing this, unless there is a good reason we should only be verifying that the state contains what we expect.